lib/temporal.explorer.ts

```diff
@@ -82,70 +81,59 @@ export class TemporalExplorer
       } as WorkerOptions;
       if (connectionOptions) {
         this.logger.verbose('Connecting to the Temporal server');
-        workerOptions.connection = await NativeConnection.connect(connectionOptions);
+        workerOptions.connection = await NativeConnection.connect(
+          connectionOptions,
+        );
       }
 
       this.logger.verbose('Creating a new Worker');
       this.worker = await Worker.create(
-        Object.assign(
-          workerOptions,
-          workerConfig,
-        ),
+        Object.assign(workerOptions, workerConfig),
       );
     }
   }
 
   getWorkerConfigOptions(name?: string): WorkerOptions {
-    return this.moduleRef.get(TEMPORAL_WORKER_CONFIG || name, {
-      strict: false,
-    });
+    return this.moduleRef.get(TEMPORAL_WORKER_CONFIG || name);
   }
 
   getNativeConnectionOptions(name?: string): NativeConnectionOptions {
-    return this.moduleRef.get(TEMPORAL_CONNECTION_CONFIG || name, {
-      strict: false,
-    });
+    return this.moduleRef.get(TEMPORAL_CONNECTION_CONFIG || name);
   }
 
   getRuntimeOptions(name?: string): RuntimeOptions {
-    return this.moduleRef.get(TEMPORAL_CORE_CONFIG || name, { strict: false });
+    return this.moduleRef.get(TEMPORAL_CORE_CONFIG || name);
+  }
+
+  getActivitiesModules(): any[] | undefined {
+    return this.moduleRef.get(TEMPORAL_ACTIVITIES_MODULES) || undefined;
   }
 
-  /**
-   *
-   * @returns
-   */
   async handleActivities() {
     const activitiesMethod = {};
 
+    const activitiesModules = this.getActivitiesModules();
     const activities: InstanceWrapper[] = this.discoveryService
       .getProviders()
-      .filter((wrapper: InstanceWrapper) =>
-        this.metadataAccessor.isActivities(
-          !wrapper.metatype || wrapper.inject
-            ? wrapper.instance?.constructor
-            : wrapper.metatype,
-        ),
+      .filter(
+        (wrapper: InstanceWrapper) =>
+          this.metadataAccessor.isActivities(
+            !wrapper.metatype || wrapper.inject
+              ? wrapper.instance?.constructor
+              : wrapper.metatype,
+          ) &&
+          (!activitiesModules || activitiesModules.includes(wrapper.metatype)),
```

Comment:
I'm open to better ideas for filtering providers solely to those provided by the module being explored. The Discovery Service is not documented. Personally, I prefer the explicit specification over the implicit discovery. I'm not a fan of "magic".